ext/php7/ip_extraction.c

```diff
@@ -446,7 +446,14 @@ static bool dd_parse_ip_address_maybe_port_pair(const char *addr, size_t addr_le
     }
     const char *colon = memchr(addr, ':', addr_len);
     if (colon) {
-        return dd_parse_ip_address(addr, colon - addr, out);
+        char *tmp_addr = safe_emalloc(addr_len, 1, 1);
+        memcpy(tmp_addr, addr, addr_len);
+        tmp_addr[addr_len] = '\0';
+        int ret = inet_pton(AF_INET6, tmp_addr, &out->v6);
```

Comment:
The code above already checks for ipv6, I'm not sure if I'm missing something but why check again?

Reply:
The code above is looking for the following pattern `[IPv6]:Port`. So given a valid `IPv6` without port, it would not enter on that conditional. Therefore, having a colon on the `addr`(on this second section) does not mean it is a `IPv4:Port`

Reply:
The alternative would be to check whether there are two colons - an IPv6 always has at least two colons, while IPv4:port is at most one.

Reply:
Thanks for pointing that out @bwoebi. I have refactor the code using this alternative